fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunnerImpl.java

```diff
@@ -311,12 +311,25 @@ public synchronized void close() {
             return;
         }
         closed = true;
+        // Ensure we call both close() methods, give preference to the transactionalRunner's exception if both throw one
+        RuntimeException caught = null;
         try {
             futureManager.close();
-        } catch (Exception e) {
-            LOGGER.warn("Caught exception while closing", e);
+        } catch (RuntimeException e) {
+            caught = e;
+        }
+        try {
+            transactionalRunner.close();
+        } catch (RuntimeException e) {
+            if (caught != null) {
+                LOGGER.warn("Caught exception while closing", caught);
+            }
+            // replace
+            caught = e;
```

Comment:
I think the modern way to do this would be to add `cause` as a suppressed exception of `e`
```suggestion
            if (caught != null) {
                caught.addSuppressed(e);
            } else {
                caught = e;
            }
```
See: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Throwable.html#addSuppressed(java.lang.Throwable)

Reply:
Yes, that would work. Done.